construi/target.py

```diff
@@ -8,29 +8,19 @@
 import random
 import sys
 import os
-
-
-def generate_build_id():
-    v = 'aeiou'
-    c = 'bdfghjklmnprstvw'
-
-    return ''.join([random.choice(v if i % 2 else c) for i in range(8)])
+import os.path
 
 
 class Target(object):
     def __init__(self, config):
         self.config = config
-        self.project = Project.from_config("construi_%s" % Target.build_id(),
+        self.project = Project.from_config(self.build_id,
                                            config.compose,
                                            docker_client(os.environ))
 
-    @classmethod
-    def build_id(cls):
-        try:
-            return cls._build_id
-        except AttributeError:
-            cls._build_id = generate_build_id()
-            return cls._build_id
+    @property
+    def build_id(self):
+        return "construi_%s" % os.path.basename(self.config.construi['working_dir'])
```

Comment:
I think everything after '%' can be moved into config and config updated so that we can write
```python
return "construi_%s" % self.config.project_name
```